src/libs/workspace-utils.ts

```diff
@@ -15,19 +15,21 @@ export const isKnownCloudProvider = (x: unknown): x is CloudProvider => {
   return (x as string) in cloudProviderTypes;
 };
 
-export interface BaseWorkspaceInfo {
+interface BaseWorkspaceInfo {
   namespace: string;
   name: string;
   workspaceId: string;
-  cloudPlatform: string;
   authorizationDomain: string[];
   createdDate: string;
   createdBy: string;
 }
 
-export type AzureWorkspaceInfo = BaseWorkspaceInfo;
+export interface AzureWorkspaceInfo extends BaseWorkspaceInfo {
+  cloudPlatform: 'Azure';
```

Comment:
Adding the values of `cloudPlatform` for each of Azure/GoogleWorkspaceInfo allows TS to narrow types.

Before:
```ts
const workspaceInfo: WorkspaceInfo = ...
if (workspaceInfo.cloudPlatform === 'Azure') {
  // workspaceInfo is typed WorkspaceInfo
} else {
  // workspaceInfo is typed WorkspaceInfo
}
```

After:
```ts
const workspaceInfo: WorkspaceInfo = ...
if (workspaceInfo.cloudPlatform === 'Azure') {
  // workspaceInfo is typed AzureWorkspaceInfo
} else {
  // workspaceInfo is typed GoogleWorkspaceInfo
}
```

Reply:
Hmm.. I'd like if we could define something like  `type CloudPlatform = 'Azure' | 'Gcp'` (the google one is probably incorrect, just psuedocode), and the some sort constant to refer to these like `cloudPlatforms.azure`. That way we have a central place to change if the API contract changes down the line instead of a bunch of magic strings scattered.

Reply:
Generally, we try to avoid directly comparing this `cloudPlatform` value to a string by instead using `isAzureWorkspace` or `isGoogleWorkspace`.

We already define
```ts
export const cloudProviderTypes: Record<CloudProvider, CloudProvider> = {
  AZURE: 'AZURE',
  GCP: 'GCP',
};
```

I'm not keen to `cloudPlatforms` when `cloudProviderTypes` already exists.

The root of the problem is that all backend services are inconsistent about the constants they use for Azure / GCP. Rawls uses 'Gcp' and 'Azure'. Leo uses 'AZURE' and 'GCP'. Data Catalog 'azure' and 'gcp'.

Reply:
Also, with this change, I'll claim that 'Gcp' and 'Azure' would no longer be magic strings since they'd be checked by TypeScript. With this change, if the API changes, TS tells us what needs to be updated. Currently, with `cloudPlatform` typed as a string, there is very little type safety.